llvm/include/llvm/Support/BranchProbability.h

```diff
@@ -62,6 +63,11 @@ class BranchProbability {
   static void normalizeProbabilities(ProbabilityIter Begin,
                                      ProbabilityIter End);
 
+  template <class ProbabilityRange>
+  static void normalizeProbabilities(ProbabilityRange &R) {
+    normalizeProbabilities(adl_begin(R), adl_end(R));
```

Comment:
I'm not sure if the adl_begin() is required here or just std::begin()/std::end() will suffice. I did what STLExtras.h seems to be doing for such range functions.

Reply:
STLExtras.h also does `&&R` for whatever reason.

Reply:
Right, I check llvm::sort() which does in place sorting and that does the same as well, so replicated it here as well.